Proxy/faraday_uart_stack/layer_4_service.py

```diff
@@ -139,8 +139,13 @@ def RxPortHasItem(self, service_number):
         try:
             return not self.receive_parsed_queue_dict[service_number].empty()
         except:
-            self.receive_service_queue_open(service_number, self.QUEUE_SIZE_DEFAULT)
-            pass
+            return None
+
+    def RxPortListOpen(self):
+        return self.receive_parsed_queue_dict.keys()
```

Comment:
It is built in the initialization of the class object:  
![image](https://cloud.githubusercontent.com/assets/339144/25073324/f580df1a-2298-11e7-8214-d54d6c30a6b6.png)

Entries are added when "opening" a port" on line 144

![image](https://cloud.githubusercontent.com/assets/339144/25073332/103f0de0-2299-11e7-8495-95ef22e63c1c.png)


Which proxy.py uses on line 66:

![image](https://cloud.githubusercontent.com/assets/339144/25073340/2cceb56e-2299-11e7-980d-c4075baf89a6.png)